services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/service/FlowCommandFactory.java

```diff
@@ -0,0 +1,345 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowhs.service;
+
+import static java.lang.String.format;
+
+import org.openkilda.floodlight.flow.request.InstallEgressRule;
+import org.openkilda.floodlight.flow.request.InstallIngressRule;
+import org.openkilda.floodlight.flow.request.InstallMultiSwitchIngressRule;
+import org.openkilda.floodlight.flow.request.InstallSingleSwitchIngressRule;
+import org.openkilda.floodlight.flow.request.InstallTransitRule;
+import org.openkilda.floodlight.flow.request.RemoveRule;
+import org.openkilda.messaging.MessageContext;
+import org.openkilda.messaging.command.switches.DeleteRulesCriteria;
+import org.openkilda.model.Flow;
+import org.openkilda.model.FlowPath;
+import org.openkilda.model.MeterId;
+import org.openkilda.model.OutputVlanType;
+import org.openkilda.model.PathId;
+import org.openkilda.model.PathSegment;
+import org.openkilda.model.SwitchId;
+import org.openkilda.model.TransitVlan;
+import org.openkilda.persistence.repositories.TransitVlanRepository;
+import org.openkilda.wfm.CommandContext;
+
+import com.fasterxml.uuid.Generators;
+import com.fasterxml.uuid.NoArgGenerator;
+import com.google.common.collect.ImmutableList;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.ListUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+public class FlowCommandFactory {
```

Comment:
Shouldn't we somehow define that this factory is tightly bound to TransitVlan encapsulation? And provide a contract for other implementation of flow command factory.